client/src/app/components/Layout/Layout.jsx

```diff
@@ -97,12 +88,7 @@ export class Layout extends PureComponent {
       viewType: newViewType
     };
 
-    this.handleSwitchView(newViewType);
-    this.setState({ pendingForViewType: true });
-
-    updateSettings(settings)
-      .catch(() => this.handleSwitchView(viewType))
-      .finally(() => this.setState({ pendingForViewType: false }));
+    this.setState({ viewType: newViewType }, saveSettings(settings));
```

Comment:
As the callback is asynchronous, please read settings again in the saveSettings callback.